slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtectionFactory.java

```diff
@@ -22,16 +23,38 @@
 
 public class SlashingProtectionFactory {
 
+  public static final int EXPECTED_DATABASE_VERSION = 6;
+  private static final String errorMsg =
+      String.format(
+          "Database does not have expected version (%s), please run migrations and try again.",
+          EXPECTED_DATABASE_VERSION);
+
   public static SlashingProtection createSlashingProtection(
       final String slashingProtectionDbUrl,
       final String slashingProtectionDbUser,
       final String slashingProtectionDbPassword) {
     final Jdbi jdbi =
         DbConnection.createConnection(
             slashingProtectionDbUrl, slashingProtectionDbUser, slashingProtectionDbPassword);
+
+    verifyVersion(jdbi);
+
     return createSlashingProtection(jdbi);
   }
 
+  private static void verifyVersion(final Jdbi jdbi) {
+    final DatabaseVersionDao databaseVersionDao = new DatabaseVersionDao();
+
+    try {
+      final Integer version = jdbi.withHandle(databaseVersionDao::findDatabaseVersion);
+      if (version.compareTo(EXPECTED_DATABASE_VERSION) != 0) {
```

Comment:
any reason not to just use equals? if (version != EXPECTED_DATABASE_VERSION)...

Reply:
what i found in our migrations is that at times, you want to add an index which is not a breaking change and can allow the service to work with a version less than the latest, while migrating to the latest in parallel. not necessarily the case here as there is not much data in the web3signer db, so you most likely won't be creating any indexes in parallel, but thought it's worth mentioning